message/message.go

```diff
@@ -282,14 +298,18 @@ func (gsr GraphSyncResponse) Status() graphsync.ResponseStatusCode { return gsr.
 // Extension returns the content for an extension on a response, or errors
 // if extension is not present
 func (gsr GraphSyncResponse) Extension(name graphsync.ExtensionName) (datamodel.Node, bool) {
-	if gsr.extensions == nil {
-		return nil, false
-	}
-	val, ok := gsr.extensions[string(name)]
-	if !ok {
-		return nil, false
+	if name == graphsync.ExtensionMetadata {
```

Comment:
I don't think metadata should be accessible as an extension

Reply:
I believe the place to put this "blend in the metadata in the extensions" is https://github.com/ipfs/go-graphsync/blob/e21664f437b58b6a5df05fe2450c081ed63162ac/message/v1/message.go#L264